src/Control/Monad/HashCons.hs

```diff
@@ -114,37 +130,56 @@ class
   -- @since 1.0.0
   refTo :: e -> m r
 
+  -- | Given a value of type @r@, fetch the cached value of type @e@.
+  --
+  -- @since 1.0.0
+  lookupRef :: r -> m e
```

Comment:
This method cannot be safe, because there is no guarantee `r` is mapped to anything. Also, what laws does this follow?

Reply:
Fixed the safety, and added the laws